platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/StopWatch.java

```diff
@@ -0,0 +1,170 @@
+ */
+public class StopWatch {
+    private final Time time;
+    private long startTime;
+    private long endTime;
+    private boolean running;
+    private boolean suspended;
+
+    /**
+     * Creates a new StopWatch.
+     */
+    public StopWatch() {
+        this(Time.getCurrent());
+    }
+
+    public StopWatch(@NonNull final Time time) {
+        this.time = Objects.requireNonNull(time);
+        this.running = false;
+        this.suspended = false;
+    }
+
+    /**
+     * Starts StopWatch
+     */
+    public void start() {
+        if (running) {
+            throw new IllegalStateException("StopWatch is already running.");
+        }
+        this.running = true;
+        this.startTime = time.nanoTime();
+    }
+
+    /**
+     * Stops StopWatch only after being running
+     */
+    public void stop() {
+        final long nanoTime = time.nanoTime();
+        if (!running || suspended) {
+            throw new IllegalStateException("StopWatch is not running.");
+        }
+        this.endTime = nanoTime;
+        this.running = false;
+    }
+
+    /**
+     * Suspends StopWatch only after being started
+     */
+    public void suspend() {
+        final long nanoTime = time.nanoTime();
+        if (!running || suspended) {
+            throw new IllegalStateException("StopWatch is not running or is already suspended.");
+        }
+        this.endTime = nanoTime;
+        this.suspended = true;
+    }
+
+    /**
+     * Resumes StopWatch only after being suspended
+     */
+    public void resume() {
+        final long nanoTime = time.nanoTime();
+        if (!suspended) {
+            throw new IllegalStateException("StopWatch is not suspended.");
+        }
+        this.startTime += (nanoTime - endTime); // Adjusting the startTime
+        this.suspended = false;
+    }
+
+    /**
+     * Returns the elapsed time in nanoseconds.
+     *
+     * @return the elapsed time in nanoseconds
+     * @throws IllegalStateException if the StopWatch has not started yet
+     */
+    public long getElapsedTimeNano() {
+        final long nanoTime = time.nanoTime();
+        if (startTime == 0 && endTime == 0) {
+            throw new IllegalStateException("StopWatch has not started yet.");
+        }
+
+        if (running) {
+            return nanoTime - startTime;
+        }
+
+        return endTime - startTime;
+    }
+
+    /**
+     * Returns if StopWatch is running
+     *
+     * @return true if StopWatch is running, false otherwise
+     */
+    public boolean isRunning() {
+        return running;
+    }
+
+    /**
+     * Returns if StopWatch is suspended
+     *
+     * @return true if StopWatch is suspended, false otherwise
+     */
+    public boolean isSuspended() {
+        return suspended;
+    }
+
+    /**
+     * Returns if StopWatch is stopped
+     *
+     * @return true if StopWatch is stopped, false otherwise
+     */
+    public boolean isStopped() {
```

Comment:
I don't think it would be obvious to the user that this method will return `true` when the stopwatch is `suspended`.

e.g. user: "Why is `isStopped()` returning true, when I never called `stop()`??"

I suggest either mentioning this behavior specifically in the doc, or modifying the method to only return `true` if the user has explicitly called `stop()`

Reply:
This is already the case, added two tests to ensure the behaviour.